src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableExtensions.cs

```diff
@@ -37,17 +37,17 @@ internal static bool TryGetCount<T>(this IEnumerable<T> sequence, out int count)
         /// <returns><c>true</c> if the count could be determined; <c>false</c> otherwise.</returns>
         internal static bool TryGetCount<T>(this IEnumerable sequence, out int count)
         {
-            var collection = sequence as ICollection;
-            if (collection != null)
+            var collectionOfT = sequence as ICollection<T>;
+            if (collectionOfT != null)
             {
-                count = collection.Count;
+                count = collectionOfT.Count;
                 return true;
             }
```

Comment:
Not necessarily related to this PR, but I couldn't help but changing the order here since the generic ICollection is by far more common than non-generic. Is this OK?